pkg/util/unstructuredhelper/write.go

```diff
@@ -0,0 +1,318 @@
+package unstructuredhelper
+
+import (
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/yaml"
+)
+
+func ValueFrom[T any](path string, from ReadMap) (T, bool) {
+	pathSplitted := strings.Split(path, ".")
+
+	// traverse to target
+	targetMap := from
+	for i := 0; i < len(pathSplitted)-1; i++ {
+		targetMap = targetMap.Map(pathSplitted[i])
+	}
+
+	// check if map has latest path
+	lastKey := pathSplitted[len(pathSplitted)-1]
+	if !targetMap.Has(lastKey) {
+		var ret T
+		return ret, false
+	}
+
+	// try to convert target value
+	targetValue, ok := targetMap[lastKey].(T)
+	if !ok {
+		var ret T
+		return ret, false
+	}
+
+	return targetValue, true
+}
+
+type TranslateFn[T any] func(in T) T
+
+func Set[T any](path string, from ReadMap, to WriteMap, translate TranslateFn[T]) {
+	pathSplitted := strings.Split(path, ".")
+
+	// traverse to target
+	targetMap := from
+	for i := 0; i < len(pathSplitted)-1; i++ {
+		targetMap = targetMap.Map(pathSplitted[i])
+	}
+
+	// check if map has latest path
+	lastKey := pathSplitted[len(pathSplitted)-1]
+
+	// to map
+	toMap := to
+	for i := 0; i < len(pathSplitted)-1; i++ {
+		toMap = toMap.Map(pathSplitted[i])
+	}
```

Comment:
nit: avoids second traversal

```suggestion
	targetMap := from
	// to map
	toMap := to
	for i := 0; i < len(pathSplitted)-1; i++ {
		targetMap = targetMap.Map(pathSplitted[i])
		toMap = toMap.Map(pathSplitted[i])
	}

	// check if map has latest path
	lastKey := pathSplitted[len(pathSplitted)-1]
```